### ISSUE TITLE
Orphan nake.bat and nake.csx files after opening VS

### ISSUE THREAD

**Issue by vansha** [yevhen/Nake#14]
Given following project structure:

```
project
|_src
| |_packages
| | |_Nake.2.3.1
| |_project.sln
| 
|_nake.bat
|_nake.csx
```

Whenever I open project.sln in Visual Studio, two sample Nake files `nake.bat` and [`nake.csx`](https://github.com/yevhen/Nake/blob/master/Build/NuGet/Nake.csx) are created inside `src` folder. Which is not desired behavior - since I have my nake files already in a parent folder.

### Analysis

That's `init.ps1` script who copies sample nake files. According to [Nuget docs](https://docs.nuget.org/create/creating-and-publishing-a-package#automatically-running-powershell-scripts-during-package-installation-and-removal) it's executed every time when solution is opened in VS.

**Comment by yevhen** [comment]
Oh, it's because you have a non-standard layout :grimacing:

I actually check whether sample Nake runner files are present in solution root or not (https://github.com/yevhen/Nake/blob/master/Build/NuGet/Init.ps1#L7). But you solution file and nake runner files are not on the same level.

Well, I prefer to leave this logic. It's really convenient when you're installing Nake first time, the runner file and sample script will be automatically created. Easy to boot novices.

Have any other ideas how can we fix this?

**Comment by vansha** [comment]
It's [pretty](https://github.com/dotnet/roslyn) [standard](https://github.com/dotnet/orleans) [layout](https://github.com/ServiceStack/ServiceStack), especially for enterprise projects :), where source code is just a part of it. On the source level we have things like database, documentation, etc. So Nake is at root directory as it deals not only with source code, while solution file in `src`.

Probably the right way of handling it is to utilize [install.ps1](https://docs.nuget.org/create/creating-and-publishing-a-package#automatically-running-powershell-scripts-during-package-installation-and-removal) script - in that case it will run only once, after first install.

But I'd rather prefer a dirty hack, that would solve my problem, and seems to not harm others. Just adding one more check :)

```csharp
$scriptDir      = split-path $script:MyInvocation.MyCommand.Path
$solutionDir    = (get-item $scriptDir ).parent.parent.parent.FullName
$mightBeRootDir = (get-item $solutionDir ).parent.FullName
$samplesDir     = Join-Path (get-item $scriptDir ).parent.FullName "sample"
$sampleCsxFile  = "Nake.csx" 
$sampleBatFile  = "Nake.bat" 

if ((Test-Path "$mightBeRootDir\Nake.csx") -or (Test-Path "$mightBeRootDir\Nake.bat")) {
    Exit
}

if ((Test-Path "$solutionDir\Nake.csx") -or (Test-Path "$solutionDir\Nake.bat")) {
    Exit
}

Write-Host "Copying sample script and runner to root folder ..." -ForegroundColor DarkYellow
Copy-Item "$samplesDir\$sampleCsxFile" $rootDir
Copy-Item "$samplesDir\$sampleBatFile" $rootDir
```

But it smells.

**Comment by yevhen** [comment]
I'm more inclined to go with Install.ps1 since Nake will be installed only once for solution.
But I need to test it. Will have a look after Easter!
Happy Easter, btw! :)

**Comment by vansha** [comment]
Happy Easter! Have a good time :)

**Comment by yevhen** [comment]
Install.ps1 won't run for solution-level packages :(
Including sample script and batch runner as content - make it project-level :(
Another solution could be is to put Readme.txt with instructions ...

**Comment by vansha** [comment]
Closing issue as it was fixed a long time ago in 2d2f5c7d8864bfd32db53f423489b14b9882d4d3 by using simple readme instead of `Install.ps1`. Thanks!
